lambdas/services/mock_pds_service.py

```diff
@@ -0,0 +1,66 @@
+import json
+
+from models.pds_models import PatientDetails
+from requests import Response
+from utils.utilities import validate_id
+from utils.exceptions import (
+    PdsErrorException,
+    PatientNotFoundException,
+    InvalidResourceIdException,
+)
+from models.pds_models import Patient
+
+
+class MockPdsApiService:
+    def __init__(self, *args, **kwargs):
+        pass
+
+    def fetch_patient_details(self, nhs_number: str) -> PatientDetails:
+        validate_id(nhs_number)
+
+        response = self.fake_pds_request(nhs_number)
+
+        if response.status_code == 200:
+            patient = Patient.model_validate(response.content)
+            patient_details = patient.get_patient_details(nhs_number)
+            return patient_details
+
+        if response.status_code == 404:
+            raise PatientNotFoundException(
+                "Patient does not exist for given NHS number"
+            )
+
+        if response.status_code == 400:
+            raise InvalidResourceIdException("Invalid NHS number")
```

Comment:
This should be caught and handled while validating the NHS number at the start of the handler. Not saying that we won't get a 400 error back from PDS, but perhaps it would be best to return the PDS response body in the Exception body, as opposed to just assuming that it's because the NHS number is invalid, I think

Reply:
Ah, this is for the fake service. Not sure if what I've written is still valid - sorry!